crates/sui-core/src/authority.rs

```diff
@@ -1952,6 +1953,20 @@ impl AuthorityState {
         }
     }
 
+    pub async fn read_table<K, V>(&self, table: &Table) -> SuiResult<BTreeMap<K, V>>
```

Comment:
We will need a bound on the number of results?

Reply:
I have changed `read_table` to `read_table_value`, which use a iterator to find the value instead of reading the whole table in mem.